File: 2023-1/Basic/bjgood01/Stack/10828.py
import sys
n = int(input())
stack = []
for i in range(n):
    s = sys.stdin.readline().strip()
    if 'push' in s:
        a = s.split()
        stack.append(a[1])
    elif 'pop' in s:
        if len(stack) != 0:  # 스택에 값이 없으면
            print(stack.pop())
        else:
            print(-1)
    elif 'empty' in s:
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    elif 'size' in s:
        print(len(stack))
    elif 'top' in s:
        if len(stack) != 0:
            print(stack[-1])
        else:
            print(-1)

File: 2023-2/Basic/yeon/stack/10828.cpp
#include <iostream>
#include <stack>

using namespace std;

int main()
{
    int t;
    cin >> t;
    
    stack <int> s;
    
    while(t--){
        
        string str;
        cin >> str;
        
        if(str == "push"){
            int tmp;
            cin >> tmp;
            s.push(tmp);
            
        } else if(str == "pop"){
            if(s.empty()) cout << -1 << '\n';
            else{
                cout << s.top() << '\n';
                s.pop();
            }
            
        } else if(str == "size"){
            cout << s.size() << '\n';
            
        } else if(str == "empty"){
            if(s.empty()) cout << 1 << '\n';
            else cout << 0 << '\n';
            
        } else if(str == "top"){
            if(s.empty()) cout << -1 << '\n';
            else cout << s.top() << '\n';
            
        }
    }

    return 0;
}
